src/popup/router/components/SwapRoute.vue

```diff
@@ -0,0 +1,155 @@
+<template>
+  <div
+    v-if="isSwapTx"
+    class="swap-route"
+  >
+    <div class="title">
+      Swap route
+    </div>
+    <span
+      v-for="swap, index of swaps"
+      :key="index"
+      class="swap"
+    >
+      <span
+        v-if="index !== 0"
+        class="divider"
+      >
+        {{ index === 1 ? '--->' : '------>' }}
+      </span>
+
+      <Tokens :tokens="[swap.from, swap.to]" />
+
+      <span
+        v-if="index !== 0 && index + 1 < swaps.length"
+        class="fee"
+      >
+        (3% pool fee)
+      </span>
+    </span>
+  </div>
+</template>
+
+<script>
+import { mapState } from 'vuex';
+import Tokens from './Tokens.vue';
+
+export default {
+  components: {
+    Tokens,
+  },
+  props: {
+    transaction: { type: Object, required: true },
+  },
+  computed: {
+    ...mapState('fungibleTokens', ['availableTokens']),
+    isSwapTx() {
+      const types = [
```

Comment:
Don't you think that `deposit` and `withdraw` should also consider a swap methods. Better would be to ask @paolomolo.